node/tessel-export.js

```diff
@@ -85,6 +94,73 @@ Tessel.prototype.close = function() {
   }, this);
 };
 
+Tessel.prototype.pwmFrequency = function(frequency, cb) {
+  if (frequency < Tessel.pwmMinFrequency || frequency > Tessel.pwmMaxFrequency) {
+    throw new Error(`${frequency} is outside the bounds of ${Tessel.pwmMinFrequency}-${Tessel.pwmMaxFrequency} Hz`);
+  }
+
+  determineDutyCycleAndPrescalar(frequency, (err, results) => {
+    if (err) {
+      throw err;
+    }
+    else {
+      Tessel.pwmPeriod = results.period;
+      Tessel.pwmPrescalar = results.prescalar;
+
+      // We are currently only using TCC Bank 0
+      // This may be expanded in the future to enable PWM on more pins
+      const TCC_ID = 0;
+
+      var packet = new Buffer(4);
+      // Write the command id first
+      packet.writeUInt8(CMD.PWM_PERIOD, 0);
+      // Write our prescalar to the top 4 bits and TCC id to the bottom 4 bits
+      packet.writeUInt8((Tessel.pwmPrescalar << 4) | TCC_ID, 1);
+      // Write our period (16 bits)
+      packet.writeUInt16BE(Tessel.pwmPeriod, 2);
+
+      // Send the packet off to the samd21
+      // on the first port object (regardless of name)
+      this.port[Object.keys(this.port)[0]].sock.write(packet, cb);
+    }
+  });
+};
+
+/*
+ Takes in a desired frequency setting and outputs the
+ necessary prescalar and duty cycle settings based on set period.
+ Outputs an object in the form of:
+ {
+  prescalar: number (0-7),
+  period: number (0-0xFFFF)
+ }
+*/
+function determineDutyCycleAndPrescalar(frequency, cb) {
+  // Maximum number of ticks before period completes
+  const MAX_PERIOD = 0xFFFF;
+  // Maximum number of unscaled ticks in a second (48 MHz)
+  const TICKS_PER_SECOND = 48000000;
+  // Current setting for the prescalar
+  var prescalarIndex = 0;
+  // Current period setting
+  var period = 0;
+
+  // If the current frequency would require a period greater than the max
+  while ((period = Math.floor((TICKS_PER_SECOND / Tessel.pwmPrescalars[prescalarIndex])/frequency)) > MAX_PERIOD) {
+    // Increase our clock prescalar
+    prescalarIndex++;
+
+    // If we have no higher prescalars
+    if (prescalarIndex === Tessel.pwmPrescalars.length) {
+      // Throw an error because this frequency is too low for our possible parameters
+      return cb(new Error('Unable to find prescalar/duty cycle parameter match for frequency'));
+    }
+  }
+
+  // We have found a period inside a suitable prescalar, return results
+  return cb(null, {period: period, prescalar: prescalarIndex});
```

Comment:
There is nothing async about this function, it should not accept a callback and should only return a value or throw an exception.